### ISSUE TITLE
Propagating IBound permission argument to classes and methods inside  

### ISSUE THREAD

**Issue by mmnosek** [Starcounter/authorization#13]
I have following situation:

Permission which is related to `Template`
```
    public class ViewTemplate : Permission
    {
        public Template Template{ get; set; }

        public ViewTemplate(Template template)
	    {
            Template = template;
	    }
    }
```
Viewmodel with permission requirement and partial class inside (without any explicit permission enforcement:
```
[RequirePermission(typeof(ViewTemplate))]
partial class TemplatePage : IBound<Template>
{
         [TemplatePage_json.TemplateSections]
        partial class SectionItem : Json, IBound<TemplateSection>
       {
       }
}
```

- When I try to open a page an error occurs: `System.Exception: Could not create check for page ChecklistDesigner.TemplatePage+SectionItemJson and permission ChecklistDesigner.Authorization.Permissions.ViewTemplate. Make sure the page is of type IBound<Template>`
- Basically permission requirement is inherited by all classes and methods inside, but parameter given to the permission is wrong. (Expects Template, but TemplateSection is given from IBound).

What do I expect?
If permission requirement is inherited, also referring argument from page above should be passed to the permission. Without that inheritance doesn't make so much sense, because at the most cases in classes inside some other data is bound to the page and I don't want to create some dummy permissions especially for every class separately.

cc @joozek78 

PS - this is HIGH priority for me :)
